### PR TITLE
Command Parser

### PR DESCRIPTION
**Pull request by user622628252416**
Regarding #15, I implemented a tree-style command parser, as well as the /help command for reference.

There is a central CommandDispatcher instance, that is contained inside of a OnceLock (pumpkin::commands::DISPATCHER), from where all commands are dispatched. There is currently no access to the level/world the command is executed in, so what a command can actually do is quite limited. This is more of a temporary solution.

### CODE REVIEW

**Comment by Snowiiii** [comment]
Hey, Mostly looks good :+1: . Could you may fix all Clippy warns ?. I also just pushed a commit which fixes the one warn in packet_encoder

File: pumpkin/src/commands/arg_player.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    None
+}
+
+/// todo: implement (so far only own name + @s is implemented)
+pub fn parse_arg_player<'a>(src: &'a mut CommandSender, arg_name: &str, consumed_args: &ConsumedArgs) -> Result<&'a mut Client, InvalidTreeError> {
+    let s = consumed_args.get(arg_name)
+        .ok_or(InvalidConsumptionError(None))?;
+
+    if let Player(client) = src {
+        if s == "@s" {
```
Comment (Bryntet) [inline pumpkin/src/commands/arg_player.rs:32]: This should probably be converted to a match statement

Comment (user622628252416) [reply pumpkin/src/commands/arg_player.rs]: Done 👍

**Comment by user622628252416** [comment]
I fixed all the warns that I could fix. I did have to suppress two dead_code warns tho because no command needing literals is implemented as of now.

**Comment by Snowiiii** [comment]
> I fixed all the warns that I could fix. I did have to suppress two dead_code warns tho because no command needing literals is implemented as of now.

Great. I just don't really like that in help command, Commands are send multiple times if they have multiple arguments. I find that very confusing. CI is btw unhappy

**Comment by Snowiiii** [comment]
Okay. Everything other looks good. Thank you :+1:

**Pull request merged**